### PR TITLE
Handles container restarts for already initialized postgresql database

### PR DESCRIPTION
**Pull request by davosian**
Fixes [issue 182](https://github.com/spring-petclinic/spring-petclinic-rest/issues/182)

### CODE REVIEW

File: src/main/resources/db/postgres/schema.sql [hunk 1]
```diff
@@ -1,102 +1,75 @@
+-- Create tables only if they don't exist
 CREATE TABLE IF NOT EXISTS vets (
-  id SERIAL,
+  id SERIAL PRIMARY KEY,
```
Comment (arey) [inline src/main/resources/db/postgres/schema.sql:3]: **question**: why did you change to this primary key? I suppose the `CONSTRAINT pk_vets PRIMARY KEY (id)` was there in order to provide the `pk_vets` name to the primary key.

File: src/main/resources/db/postgres/schema.sql [hunk 2]
```diff
@@ -1,102 +1,75 @@
+-- Create tables only if they don't exist
```
Comment (arey) [inline src/main/resources/db/postgres/schema.sql:1]: **suggestion**: we could maybe refactor and modernize the PostgreSQL script following the work from the canonical version of Spring Petclinic: https://github.com/spring-projects/spring-petclinic/blob/main/src/main/resources/db/postgres/schema.sql and https://github.com/spring-projects/spring-petclinic/blob/main/src/main/resources/db/postgres/data.sql 
We should bear in mind that we support 3 DAO implementations

File: src/main/resources/db/postgres/schema.sql [hunk 3]
```diff
@@ -1,102 +1,75 @@
...
+  id SERIAL PRIMARY KEY,
   pet_id INT NOT NULL,
   visit_date DATE,
   description VARCHAR(255),
-  FOREIGN KEY (pet_id) REFERENCES pets(id),
-  CONSTRAINT pk_visits PRIMARY KEY (id)
+  FOREIGN KEY (pet_id) REFERENCES pets(id)
 );
 
-ALTER SEQUENCE visits_id_seq RESTART WITH 100;
```
Comment (arey) [inline src/main/resources/db/postgres/schema.sql:84]: **question**: I'm not aware of the potential side effects of removing the sequence reset. Did you?

File: src/main/resources/db/postgres/schema.sql (hunk 1 above)
Comment (davosian) [reply src/main/resources/db/postgres/schema.sql:3]: You are correct. This should be reverted.

File: src/main/resources/db/postgres/schema.sql (hunk 2 above)
Comment (davosian) [reply src/main/resources/db/postgres/schema.sql:1]: Being closer to the canonical version certainly makes sense to me.

File: src/main/resources/db/postgres/schema.sql (hunk 3 above)
Comment (davosian) [reply src/main/resources/db/postgres/schema.sql:84]: Upon further testing, I found that the code currently submitted with all the changes does not work correctly: creating new objects fails. I did not check yet whether this is related to the primary key changes or if it has to do with the sequence reset. The error I am getting is of type "primary key exists" when inserting into the database.

Considering a major rework by aligning with the canonical version probably makes the most sense. I was also wondering whether the same issues exist for using the hql and mysql versions or whether this is specific to postgresql.

Comment (arey) [reply src/main/resources/db/postgres/schema.sql:84]: Thank you @davosian for your tests.
Rewriting the PostgreSQL (and probably MySQL) DDL and DML scripts to be idempotent seems like a good idea. @dsyer did it on the canonical version.
Would you like to contribute to this extension?

HSQLDB and H2 are used as in-memory databases. Schema and data are re-created each time the application is started. So we don't need to rework them.

Comment (davosian) [reply src/main/resources/db/postgres/schema.sql:84]: Hi @arey , unfortunately I currently do not have the capacity to move forward with more involved contributions (besides being far from a Spring expert). I'd be happy to test any improvements with my current setup, though. I mainly set up a docker compose stack with postgresql, petclinic rest and petclinic angular.

**Comment by arey** [comment]
Close in favour of https://github.com/spring-petclinic/spring-petclinic-rest/pull/185
